odl/deform/linear_deformation.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Shape-based reconstruction using optimal information transportation.
+
+The Fisher-Rao metric is used in regularization term. And L2 data matching
+term is used in fitting term.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+from odl.operator import Operator
+from odl.discr import DiscreteLp, Divergence
+from odl.space import ProductSpace
+
+__all__ = ('LinearDeformation', 'MassPreservingLinearDeformation')
+
+
+class LinearDeformation(Operator):
+    """Linearized deformation."""
+    def __init__(self, offsets=None):
```

Comment:
A candidate for common code would be
```python
class LinearizedDeformationOperatorBase:
    def deform(self, template, deformation):
        # deform template with deformation (given in a certain way)
```

What else could there be? Derivatives and adjoints require too much internal knowledge of the concrete operator to be abstracted out completely, as far as I see it. However, the product space guy could easily use the derivatives and their adjoints from the "partial" operators.